store/store.go

```diff
@@ -1207,8 +1207,12 @@ func (s *Store) WriteCatalogs(names io.Writer) error {
 		Accept: halJsonContentType,
 	}
 
+	// do not log body for catalog updates (its huge)
+	client := httputil.NewHTTPClient(&httputil.ClientOpts{
+		MayLogBody: false,
```

Comment:
Similar code in `store.New()` also sets a `Timeout` field.

Reply:
Maybe an idea for deeper refactoring, but request context looks like a perfect place to enable/disable logging on demand. The `*Request` will reach the `RoundTripper` where you could check if logging is enabled or not. You could then refactor the code:
```
package httputil

...

type httputilCtxKeyType int
var logBodyKey httputilCtxKeyType = 1

func WithLoggedBody(ctx context.Context) context.Context {
    return context.WithValue(ctx, logBodyKey, true)
}

func WithoutLoggedBody(ctx context.Context) context.Context {
    return context.WithValue(ctx, logBodyKey, false)
}

// Returns the log body flag. Additional returned parameter indicates if the flag was set.
// NOTE: or maybe make the log flag tristate: on, off, unset
func isLogBodyEnabled(ctx context.Context) (bool, bool) {
    if v, ok := ctx.Value(logBodyKey).(bool); ok {
        return v, ok
    }
    return false, false
}
... 
func (tr *LoggedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
	dump := tr.body
	enabled, wasSet := isLogBodyEnabled(req.Context())
	if wasSet {
		dump = dump & enabled
	}
	flags := tr.getFlags()

	if flags.debugRequest() {
		buf, _ := httputil.DumpRequestOut(req, (dump && flags.debugBody())
		logger.Debugf("> %q", buf)
	}

	rsp, err := tr.Transport.RoundTrip(req)

	if err == nil && flags.debugResponse() {
		buf, _ := httputil.DumpResponse(rsp, dump && flags.debugBody())
		logger.Debugf("< %q", buf)
	}

	return rsp, err
}

// actual call site
func (s *Store) WriteCatalogs(..) {
	...
	doRequest := func() (*http.Response, error) {
		
		return s.doRequest(WithLoggedBody(context.TODO()), s.client, reqOptions, nil)
 	}

}
```

Reply:
agreed, we need to sit down sometime and actually switch to context.

Reply:
I played with this a bit, but unfortunately its a bit tricky with go-1.6 (which we use). We don't have `req.Context()` here so in the roundtripper we have no easy access to the context here.

Reply:
Bummer 😞 forgot that we need to support 1.6 still.

Reply:
we could stick a noLogClient on the side of the current client in Store to reuse